lib/repository/readonly-repository.ts

```diff
@@ -452,7 +452,8 @@ export class ReadonlyRepository<TEntity extends ModelBase, TRawData = any> imple
 			model$ = of(model);
 		}
 
-		return entity.readonly ? model$.pipe(map(model => Object.freeze(model))) : model$;
+		// typecasting is used here until bug in TS is fixed (https://github.com/Microsoft/TypeScript/issues/26495)
+		return (entity.readonly ? model$.pipe(map(model => Object.freeze(model))) : model$) as TEntity extends TConcreteEntity ? Observable<TEntity> : Observable<TConcreteEntity>;
```

Comment:
Seems like two separate issues. Shouldn't they be two separate PRs?

Reply:
They are. See #6. It'll be completed before this one.

Reply:
#6 completed, this change is no longer part of this PR (it was added just so I can build this and test it out).